components/TextFields/styling/README.md

```diff
@@ -0,0 +1,77 @@
+#  Text field styles
+
+MDC's text fields are designed to be styled by controllers implementing the `MDCTextInputController` protocol. The following controllers are included in the text field component and each one is highly customizeable to express your app's branding and styling.
+
+Some of the classes are marked as "Preferred". These are known, thru extensive UX research, to have big performance gains in usability.
+
+Another two of the classes are "Legacy". They are included to support older designs and avoid breaking changes that come from the size of newer styles.
+
+## MDCTextInputControllerOutlined (Preferred)
+
+![MDCTextInputControllerOutlined](docs/assets/MDCTextInputControllerOutlined.gif)
+
+- Stroked border
+- Border-crossing, floating placeholder
+- No fill
+
+### Notes:
+This class is intended for single line text fields only. For a visually compatible multi-line controller, use `MDCTextInputControllerOutlinedTextArea`.
+
+## MDCTextInputControllerFilled (Preferred)
+
+![MDCTextInputControllerFilled](docs/assets/MDCTextInputControllerFilled.gif)
+
+- Optionally stroked border / underline
+- Floating placeholder
+- Colored fill
+
+## MDCTextInputControllerOutlinedTextArea (Preferred)
+
+- Stroked border
+- Border-crossing, floating placeholder
+- No fill
+
+### Notes:
+This class is intended for multi-line text fields only. For a visually compatible single line controller, use `MDCTextInputControllerOutlined`.
+
+## MDCTextInputControllerDefault
+
+![MDCTextInputControllerDefault](docs/assets/MDCTextInputControllerDefault.gif)
+
+- Underline
+- Optionally floating placeholder (default is to float)
```

Comment:
Grammar nit: "Optional floating placeholder" instead of "optionally"

Reply:
Done.